language/move-stdlib/Cargo.toml

```diff
@@ -18,13 +18,13 @@ diem-workspace-hack = { path = "../../common/workspace-hack" }
 move-vm-types = { path = "../move-vm/types" }
 move-binary-format = { path = "../move-binary-format" }
 move-core-types = { path = "../move-core/types" }
-diem-crypto = { path = "../../crypto/crypto" }
 move-vm-runtime = { path = "../move-vm/runtime" }
 
 log = "0.4.14"
 walkdir = "2.3.1"
 smallvec = "1.6.1"
 sha2 = "0.9.3"
+sha3 = "0.9.1"
```

Comment:
I think we should also benchmark this change before we apply it. The underlying hash library in `diem-crypto` is `tiny_keccak`, not `sha3`, the reason being `tiny_keccak` was slightly faster back then and this might still be true. Some benchmarks are available here 
https://github.com/debris/tiny-keccak#benchmarks.
One option is to use `tiny_keccak` vs `sha3` if the benchmarks favor the first if the same time we want to remove the diem dependency.

Reply:
It looks like now they are almost equally fast:
```
test rust_crypto_sha3_256_input_32_bytes   ... bench:         597 ns/iter (+/- 85) = 53 MB/s
test rust_crypto_sha3_256_input_4096_bytes ... bench:      17,690 ns/iter (+/- 780) = 231 MB/s
test tiny_keccak_sha3_256_input_32_bytes   ... bench:         602 ns/iter (+/- 71) = 53 MB/s
test tiny_keccak_sha3_256_input_4096_bytes ... bench:      18,052 ns/iter (+/- 792) = 226 MB/s

test rust_crypto_sha3_256_input_32_bytes   ... bench:         636 ns/iter (+/- 47) = 50 MB/s
test rust_crypto_sha3_256_input_4096_bytes ... bench:      18,700 ns/iter (+/- 2,725) = 219 MB/s
test tiny_keccak_sha3_256_input_32_bytes   ... bench:         632 ns/iter (+/- 49) = 50 MB/s
test tiny_keccak_sha3_256_input_4096_bytes ... bench:      17,897 ns/iter (+/- 763) = 228 MB/s

test rust_crypto_sha3_256_input_32_bytes   ... bench:         615 ns/iter (+/- 4) = 52 MB/s
test rust_crypto_sha3_256_input_4096_bytes ... bench:      17,608 ns/iter (+/- 775) = 232 MB/s
test tiny_keccak_sha3_256_input_32_bytes   ... bench:         599 ns/iter (+/- 24) = 53 MB/s
test tiny_keccak_sha3_256_input_4096_bytes ... bench:      19,040 ns/iter (+/- 1,418) = 215 MB/s

test rust_crypto_sha3_256_input_32_bytes   ... bench:         595 ns/iter (+/- 23) = 53 MB/s
test rust_crypto_sha3_256_input_4096_bytes ... bench:      17,665 ns/iter (+/- 794) = 231 MB/s
test tiny_keccak_sha3_256_input_32_bytes   ... bench:         612 ns/iter (+/- 34) = 52 MB/s
test tiny_keccak_sha3_256_input_4096_bytes ... bench:      18,328 ns/iter (+/- 796) = 223 MB/s

test rust_crypto_sha3_256_input_32_bytes   ... bench:         598 ns/iter (+/- 24) = 53 MB/s
test rust_crypto_sha3_256_input_4096_bytes ... bench:      17,797 ns/iter (+/- 901) = 230 MB/s
test tiny_keccak_sha3_256_input_32_bytes   ... bench:         599 ns/iter (+/- 39) = 53 MB/s
test tiny_keccak_sha3_256_input_4096_bytes ... bench:      17,936 ns/iter (+/- 806) = 228 MB/s

test rust_crypto_sha3_256_input_32_bytes   ... bench:         596 ns/iter (+/- 39) = 53 MB/s
test rust_crypto_sha3_256_input_4096_bytes ... bench:      18,086 ns/iter (+/- 903) = 226 MB/s
test tiny_keccak_sha3_256_input_32_bytes   ... bench:         609 ns/iter (+/- 43) = 52 MB/s
test tiny_keccak_sha3_256_input_4096_bytes ... bench:      17,944 ns/iter (+/- 711) = 228 MB/s

test rust_crypto_sha3_256_input_32_bytes   ... bench:         594 ns/iter (+/- 43) = 53 MB/s
test rust_crypto_sha3_256_input_4096_bytes ... bench:      18,150 ns/iter (+/- 1,641) = 225 MB/s
test tiny_keccak_sha3_256_input_32_bytes   ... bench:         633 ns/iter (+/- 87) = 50 MB/s
test tiny_keccak_sha3_256_input_4096_bytes ... bench:      19,486 ns/iter (+/- 1,074) = 210 MB/s

test rust_crypto_sha3_256_input_32_bytes   ... bench:         614 ns/iter (+/- 4) = 52 MB/s
test rust_crypto_sha3_256_input_4096_bytes ... bench:      17,926 ns/iter (+/- 637) = 228 MB/s
test tiny_keccak_sha3_256_input_32_bytes   ... bench:         599 ns/iter (+/- 82) = 53 MB/s
test tiny_keccak_sha3_256_input_4096_bytes ... bench:      18,856 ns/iter (+/- 1,376) = 217 MB/s

test rust_crypto_sha3_256_input_32_bytes   ... bench:         604 ns/iter (+/- 42) = 52 MB/s
test rust_crypto_sha3_256_input_4096_bytes ... bench:      17,543 ns/iter (+/- 642) = 233 MB/s
test tiny_keccak_sha3_256_input_32_bytes   ... bench:         599 ns/iter (+/- 51) = 53 MB/s
test tiny_keccak_sha3_256_input_4096_bytes ... bench:      18,699 ns/iter (+/- 2,813) = 219 MB/s
```
I assume this means I can keep using `sha3`? Any other reason I should still prefer `tiny_keccak `?

Reply:
Great, we're good to go